### PR TITLE
LUIS runtime changes

### PR DESCRIPTION
**Pull request by diberry**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by diberry** [comment]
Authoring changes in different PR https://github.com/Azure/azure-rest-api-specs/pull/5416

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5510 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5510/commits/8284ab9e856f9724c60ca00d325cff8166a6a7d9

File: specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json [hunk 1]
```diff
@@ -5,11 +5,52 @@
...
   "x-ms-parameterized-host": {
-    "hostTemplate": "{Endpoint}/luis/v2.0",
-    "useSchemePrefix": false,
+    "hostTemplate": "{AzureRegion}.api.cognitive.microsoft.{AzureCloud}/luis/v2.0",
     "parameters": [
       {
-        "$ref": "#/parameters/Endpoint"
+        "name": "AzureRegion",
+        "description": "Supported Azure regions for Cognitive Services endpoints",
+        "x-ms-parameter-location": "client",
```
Comment (dsgouda) [inline specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: Is there a reason these are supposed to be client level parameters?

**Review by dsgouda — COMMENTED** [review]
Looks fine, posted a question

File: specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json (hunk 1 above)
Comment (diberry) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @nebadr @AmirGeorge - client level params?

File: specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json [hunk 2]
```diff
@@ -5,11 +5,52 @@
     "version": "2.0"
   },
   "x-ms-parameterized-host": {
-    "hostTemplate": "{Endpoint}/luis/v2.0",
-    "useSchemePrefix": false,
+    "hostTemplate": "{AzureRegion}.api.cognitive.microsoft.{AzureCloud}/luis/v2.0",
```
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: This change will prevent the users from using customized domain/containers.
Please revert this change.

If you believe this change is a must, we can kickstart an email thread to discuss abt this.

**Review by yangyuan — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json (hunk 1 above)
Comment (AmirGeorge) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @tsuwandy for possible input.

**Comment by dsgouda** [comment]
startbuild

File: specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json (hunk 1 above)
Comment (nebadr) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @dsgouda We are not aware exactly why but is it causing problem?

Comment (dsgouda) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: This is not necessarily a problem. We want to ensure your intent is to tie each client instance to a particular Azure instance.
Please confirm the same.

Comment (nebadr) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @dsgouda Yes, this is our intent. Confirmed.

File: specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json (hunk 2 above)
Comment (dsgouda) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @diberry has this been addressed?

Comment (diberry) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @nebadr @AmirGeorge Is this something you can address? I don't think I introduced this issue.

Comment (nebadr) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @dsgouda @diberry this change has been reverted in the second commit.

Comment (diberry) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @nebadr - should I close this PR?

Comment (nebadr) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @diberry We want to merge this PR. I meant the change has been reverted here in this PR but in the second commit (this change appears to be outdated).

Comment (diberry) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @nebadr Do I need to pull something into this PR to fix or it is good to go as it is now?

Comment (nebadr) [reply specification/cognitiveservices/data-plane/LUIS/Runtime/stable/v2.0/LUIS-Runtime.json]: @diberry No, it is good. Thanks.

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@yangyuan please approve the PR so I can merge it.

**Comment by dsgouda** [comment]
@yangyuan Please approve so we can merge this PR

**Review by yangyuan — APPROVED** [review]

**Comment by dsgouda** [comment]
@sarangan12 could you take a look at the Ruby failures?

**Comment by dsgouda** [comment]
@diberry would you like the auto generated Net Sdk PR [here](https://github.com/Azure/azure-sdk-for-net/pull/5510) to be merged or are you planning to manually open one?

**Comment by diberry** [comment]
@dsgouda Please auto-generate the .Net SDK.
